dali/base/sysinfologger.cpp

```diff
@@ -0,0 +1,891 @@
+                    }
+                }
+            }
+        }
+        if (matchEndYear)
+        {
+            if (tyear>matchEndYear)
+                return false;
+            if (tyear==matchEndYear)
+            {
+                if (matchEndMonth)
+                {
+                    if (tmonth>matchEndMonth)
+                        return false;
+                    if (tmonth==matchEndMonth)
+                    {
+                        if (matchEndDay && tday>matchEndDay)
+                            return false;
+                    }
+                }
+            }
+        }
+        return true;
+    }
+    virtual bool queryHiddenOnly() const override
+    {
+        return hiddenOnly;
+    }
+    virtual bool queryVisibleOnly() const override
+    {
+        return visibleOnly;
+    }
+    virtual unsigned __int64 queryMatchTimeStamp() const override
+    {
+        return matchTimeStamp;
+    }
+    virtual unsigned queryStartYear() const override
+    {
+        return matchStartYear;
+    }
+    virtual unsigned queryStartMonth() const override
+    {
+        return matchStartMonth;
+    }
+    virtual unsigned queryStartDay() const override
+    {
+        return matchStartDay;
+    }
+    virtual unsigned queryEndYear() const override
+    {
+        return matchEndYear;
+    }
+    virtual unsigned queryEndMonth() const override
+    {
+        return matchEndMonth;
+    }
+    virtual unsigned queryEndDay() const override
+    {
+        return matchEndDay;
+    }
+    virtual unsigned queryMatchYear() const override
+    {
+        return (matchStartYear==matchEndYear ? matchStartYear : 0);
+    }
+    virtual unsigned queryMatchMonth() const override
+    {
+        return (matchStartMonth==matchEndMonth ? matchStartMonth : 0);
+    }
+    virtual unsigned queryMatchDay() const override
+    {
+        return (matchStartDay==matchEndDay ? matchStartDay : 0);
+    }
+    virtual const char * queryMatchSource() const override
+    {
+        return matchSource.str();
+    }
+    virtual LogMsgCode queryMatchCode() const override
+    {
+        return matchCode;
+    }
+    virtual LogMsgAudience queryMatchAudience() const override
+    {
+        return matchAudience;
+    }
+    virtual LogMsgClass queryMatchClass() const override
+    {
+        return matchClass;
+    };
+    virtual unsigned __int64 queryMatchMsgId() const override
+    {
+        return matchMsgId;
+    };
+
+};
+
+ISysInfoLoggerMsgFilter * createSysInfoLoggerMsgFilter()
+{
+    return new CSysInfoLoggerMsgFilter();
+}
+
+class CSysInfoLoggerMsgIterator : public CInterface, implements ISysInfoLoggerMsgIterator
+{
+    Linked<ISysInfoLoggerMsgFilter> filter;
+    Owned<IRemoteConnection> conn = nullptr;
+    bool updateable = false;
+    Owned<IPropertyTree> root = nullptr;
+    Owned<IPropertyTreeIterator> msgIter = nullptr;
+    CSysInfoLoggerMsg infoMsg;
+
+    bool ensureMatch()
+    {
+        if (filter->hasDateRange())
```

Comment:
I have modified hasDateRange so that it returns true only if there is a date range and not when there is a single date match, so this doesn't have to change.